### ISSUE TITLE
Fix/workaround PyTorch types that don't serialize 

### ISSUE THREAD

**Issue by mbohlkeschneider** [awslabs/gluon-ts#1193]
## Description
Some PyTorch types don't serialize properly using [`encode`](https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/core/serde/_base.py#L81). For example, [this test](https://github.com/awslabs/gluon-ts/blob/master/test/torch/model/test_torch_predictor.py#L50) fails if it is changed to this, which would be the proper typing of the arguments.

```
    predictor = PyTorchPredictor(
        prediction_length=prediction_length,
        freq="1H",
        input_names=["past_target"],
        prediction_net=pred_net,
        batch_size=16,
        input_transform=transformation,
        device=torch.device('cpu'),
    )
```
 We should either fix the encoding or bypass this.

**Comment by lostella** [comment]
I haven't tried this, but I suspect this might be a fix:
1. implement a handler to let `validate` know that it shouldn't try encoding a `device` object, similar to the [this one for MXNet's `ParameterDict`](https://github.com/awslabs/gluon-ts/blob/7187e25eda1380d8665f83f6d4c15ec78b8c745c/src/gluonts/mx/component.py#L131-L133);
2. set the device upon de-serialization (i.e. take it as optional input to the `deserialize` method).

**Comment by mbohlkeschneider** [comment]
Device is a special case in some way because you might have them on different devices. But in general, you want the content to be the same (there is no point in changing the loss function upon inference). I would rather maybe construct the object internally via a factory mechanism and pass a string to the constructor which should work well with serialization.
